libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgress.kt

```diff
@@ -56,7 +56,7 @@ data class RouteProgress(
      * Builder for [RouteProgress]
      */
     class Builder {
-        private var directionsRoute: DirectionsRoute? = null
+        private var directionsRoute: DirectionsRoute = DirectionsRoute.builder().build()
```

Comment:
Why not require this in the builder's constructor, or throw an exception when calling `build` and the route was not provided?

Reply:
That's another option. Wondering though if it'd have any side effects, like unexpected runtime crashes 🤔 We're currently handling the "empty" state as no route, so we should be good. Am I missing something? `RouteProgress` is "read-only" from a client point of view, we create it internally so thinking if requiring a route through the constructor (which BTW could be empty) would add value here 🤔